src/esp/assets/ResourceManager.cpp

```diff
@@ -945,14 +953,20 @@ void ResourceManager::loadMeshes(Importer& importer,
     auto* gltfMeshData = static_cast<GltfMeshData*>(currentMesh.get());
     gltfMeshData->setMeshData(importer, iMesh);
 
+    // compute the mesh bounding box regardless of shifting
+    gltfMeshData->BB = computeMeshBB(gltfMeshData);
+
     // see if the mesh needs to be shifted
     if (shiftOrigin) {
       // compute BB center if necessary ([0,0,0])
-      if (offset[0] == 0 && offset[1] == 0 && offset[2] == 0)
-        offset = -computeMeshBBCenter(gltfMeshData);
-      // translate the mesh if necessary
-      if (!(offset[0] == 0 && offset[1] == 0 && offset[2] == 0))
+      if (offset[0] == 0 && offset[1] == 0 && offset[2] == 0) {
+        offset = -gltfMeshData->BB.center();
```

Comment:
You are correct, all meshes get the same translation currently. If we are transforming mesh hierarchies, we want all of them transformed together so that later their transforms will relate correctly. Really this shifting was intended to align single object mesh origins with their centers of mass so we could directly consume those meshes to create collision objects. This will not work properly for mesh hierarchies and we need a larger PR to address this down the line. I'll create an issue for this for now.

Reply:
Opened an issue for this: #301